### PR TITLE
[CI] Integration tests matrix pipeline

### PR DESCRIPTION
**Pull request by pazone**
## What does this PR do?
Needs https://github.com/elastic/elastic-agent/pull/4957.

Integration testing matrix pipeline that will run integration tests for all supported platforms.
The pipeline will be triggered every Thursday at 00:00. Or manually from buildkite. The pipeline won't be triggered by any github events.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Related issues

-

## Questions to ask yourself

- How are we going to support this in production?
- How are we going to measure its adoption?
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

**Review by blakerouse — APPROVED** [review]
Looks good.

**Comment by pierrehilbert** [comment]
@pazone are we good to merge here?

**Comment by pazone** [comment]
@pierrehilbert Not yet. I'd like improve one thing

**Review by alexsapran — APPROVED** [review]

File: .buildkite/pipeline.integration-test-matrix.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+
+env:
+  DOCKER_REGISTRY: "docker.elastic.co"
+  VAULT_PATH: "kv/ci-shared/observability-ingest/cloud/gcp"
```
Comment (v1v) [inline .buildkite/pipeline.integration-test-matrix.yml:5]: What is the purpose of this environment variable? I cannot find any references to it in this PR.

**Pull request merged**